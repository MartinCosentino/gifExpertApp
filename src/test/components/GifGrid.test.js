import React from 'react';
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');
const { shallow } = require("enzyme");
const { GifGrid } = require("../../components/GifGrid");




describe('Pruebas sobre el GifGrid', () => {

    const category = 'Blue Exorcist';

    test('Debe de coincidir con el matchsnapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper).toMatchSnapshot();


    });

    test('debe de mostrar items cuando se cargan imagenes con useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.png',
            title: 'Cualquier Cosa'
        }, {
            id: '123',
            url: 'https://localhost/cualquier/cosa.png',
            title: 'Cualquier Cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        const wrapper = shallow(<GifGrid category={category} />);

        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });


});
