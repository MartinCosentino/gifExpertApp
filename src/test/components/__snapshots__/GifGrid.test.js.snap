// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el GifGrid Debe de coincidir con el matchsnapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInLeft"
  >
    Blue Exorcist
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas sobre el GifGrid debe de mostrar items cuando se cargan imagenes con useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__backInLeft"
  >
    Blue Exorcist
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier Cosa"
      url="https://localhost/cualquier/cosa.png"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier Cosa"
      url="https://localhost/cualquier/cosa.png"
    />
  </div>
</Fragment>
`;
